### PR TITLE
[UNDERTOW-2149] Add path to cookie lookup

### PR DESCRIPTION
**Pull request by baranowb**
Jira: https://issues.redhat.com/browse/UNDERTOW-2149
depends on #1366

### CODE REVIEW

File: core/src/main/java/io/undertow/server/HttpServerExchange.java [hunk 1]
```diff
@@ -1184,18 +1186,92 @@ public HttpServerExchange setRequestCookie(final Cookie cookie) {
     }
 
     public Cookie getRequestCookie(final String name) {
+        return getRequestCookie(name,null);
```
Comment (fl4via) [inline core/src/main/java/io/undertow/server/HttpServerExchange.java:1189]: missing blank space between name, and null

File: core/src/main/java/io/undertow/server/HttpServerExchange.java [hunk 2]
```diff
@@ -1184,18 +1186,92 @@ public HttpServerExchange setRequestCookie(final Cookie cookie) {
...
+            if(!requestPath.startsWith(path)) {
+                return null;
+            }
+        }
+        Cookie potentialReturn = null;
+        for (final Cookie c : optionsToMatch) {
+            if (cookiePathMatch(c, path) && cookieDomainMatch(c)) {
+                if(potentialReturn == null) {
+                    potentialReturn = c;
+                    //if(true) return potentialReturn;
```
Comment (fl4via) [inline core/src/main/java/io/undertow/server/HttpServerExchange.java:1228]: probably this line should be deleted

File: core/src/main/java/io/undertow/server/HttpServerExchange.java [hunk 3]
```diff
@@ -1184,18 +1186,92 @@ public HttpServerExchange setRequestCookie(final Cookie cookie) {
...
+                } else {
+                    // at this point length of path is a factor only ?
```
Comment (fl4via) [inline core/src/main/java/io/undertow/server/HttpServerExchange.java:1230]: agreed

File: core/src/main/java/io/undertow/server/HttpServerExchange.java [hunk 4]
```diff
@@ -1184,18 +1186,92 @@ public HttpServerExchange setRequestCookie(final Cookie cookie) {
...
+        return potentialReturn;
     }
 
+    private String computeCookiePath(final Cookie c) {
+        //NOTE: this is shady, seems like default is used if no value is present, though UAs should not send cookies to wrong headers:
+        //5.2 https://www.ietf.org/rfc/rfc2109.txt
+        if (c.getPath() != null) {
+            return c.getPath();
+        } else {
+            // NOTE: assume no Path == '/' or compute default-path like UAs?
```
Comment (fl4via) [inline core/src/main/java/io/undertow/server/HttpServerExchange.java:1251]: leave it like this for now, this is another shady area

**Review by fl4via — CHANGES_REQUESTED** [review]
Just small changes, overall it looks good

File: core/src/main/java/io/undertow/server/HttpServerExchange.java (hunk 2 above)
Comment (baranowb) [reply core/src/main/java/io/undertow/server/HttpServerExchange.java:1228]: probably!

File: core/src/main/java/io/undertow/server/HttpServerExchange.java [hunk 5]
```diff
@@ -1184,18 +1186,92 @@ public HttpServerExchange setRequestCookie(final Cookie cookie) {
...
+        //NOTE: https://www.rfc-editor.org/rfc/rfc6265#section-5.1.4
+        //NOTE: https://www.rfc-editor.org/rfc/rfc6265#section-5.1.3
+        // NOTE: Its possible to receive request "/sub" with cookies "/" and "/sub" -
+        //we should return longest match if path is not directly specified
+        final String requestPath = this.getRequestPath();
+        if(path == null) {
+            path = requestPath;
+        } else {
+            //path must be a prefix or match request
+            if(!requestPath.startsWith(path)) {
```
[the remark this answers is not in the document]
Comment (baranowb) [reply core/src/main/java/io/undertow/server/HttpServerExchange.java:1219]: [1265](https://github.com/undertow-io/undertow/pull/1380/commits/c68453ead28f7de0dfe299204b252a6b2a0788cd#diff-a12450d8275afb8541a4d3f79f04e2439ecb7195dfff1af46cce1bd0e2e73c31R1265) - does check for this, unless I miss something?

**Pull request closed without merging**